src/arch.rs

```diff
@@ -263,7 +250,4 @@ pub enum SingleStepGdbBehavior {
     ///
     /// e.g: MIPS
     Ignored,
-    /// Unknown behavior - no one has tested this platform yet. If possible,
-    /// please conduct a test + upstream your findings to `gdbstub_arch`.
-    Unknown,
```

Comment:
Removing this variant would technically be a breaking API change, and given how "small" the change is, I'd rather not push out a whole new gdbstub 0.7 for that.

Instead, lets just mark this variant as `#[doc(hidden)]`, and then I'll spin up a tracking task to remove the variant entirely whenever the next breaking release comes out.

Reply:
Why you need to release `gdbstub` 0.7 only for this?
You can just leave this commit in the `dev/0.7` branch and release it along with all the other changes when they are done in `gdbstub` 0.7.
And you can release `gdbstub_arch` 0.2.1 for this commit. Then people won't encounter breaking API change for this even `gdbstub` 0.7 is released in the future.

Reply:
OK, I add `SingleStepGdbBehavior::Unknown` back, although I'm really reluctant to do this.
I think it's really not necessary to push out `gdbstub` 0.6.1 that the only difference with 0.6.0 is `#[doc(hidden)]`, because even if `Unknown` exists, nobody will use it. People won't notice it if they use the defined architecture, and if they want to create a new architecture, they will clear it is not something they need. You only need to push out `gdbstub_arch` 0.2.1.

> That said, given that these changes would all be non-breaking and land in `gdbstub_arch`, it's fine to have some `Unknown`s for now and update things as folks have time to look into each arch.

So the first commit after `gdbstub` 0.6.0 is a breaking API change🙃
* * *
Fine, it's your project and you are always right. You can just revert the later commit after you create the `dev/0.7` branch🙂

Reply:
When I implemented the Unknown variant, I knew full well that it'd end up getting removed fairly quickly, and that I'd need to push out a new minor `gdbstub_arch` release, along with (possibly) a new minor `gdbstub` release that soft-deprecates the newly added variant.

While it may seem silly to add a feature to 0.6 that immediately gets soft deprecated, the reality is that - for my own sanity - I really just wanted to get 0.6 out the door ¯\\\_(ツ)_/¯

> So the first commit after `gdbstub` 0.6.0 is a breaking API change🙃

Not super sure what you mean by that.

Of course, whatever first commit ends up in `dev/0.7` will be a breaking change. That's the whole point of opening a new `dev` branch - to put in any new breaking changes.

It could have just as easily been the case that you disappeared for a few months, and in the meantime, someone else came along and opened a PR to add a new non-breaking protocol extension to the library, which wouldn't even require opening a new `dev/0.7` branch